sale_timesheet_line_exclude/models/account_analytic_line.py

```diff
@@ -29,26 +29,24 @@ def _onchange_exclude_from_sale_order(self):
 
     @api.constrains("exclude_from_sale_order")
     def _constrains_exclude_from_sale_order(self):
-        if self.filtered(
-            lambda line: line.timesheet_invoice_id
-            and line.so_line.product_id.invoice_policy == "delivery"
-        ):
-            raise ValidationError(
-                _(
-                    "You can not modify timesheets in a way that would affect"
-                    " invoices since these timesheets were already invoiced."
+        for line in self:
```

Comment:
Optimization?

Reply:
Yup, removed one for loop :grin:

Reply:
Whoops, I mean skipping other elements instead of continuing the for loop